scipy/signal/tests/test_cont2discrete.py

```diff
@@ -368,3 +416,41 @@ def test_c2d_tf(self):
         # Somehow a lot of numerical errors
         assert_allclose(sys.den, den_res, atol=0.02)
         assert_allclose(sys.num, num_res, atol=0.02)
+
+
+class TestC2dInvariants:
+    # Some test cases for checking the invariances.
+    # Array of triplets: (system, sample time, number of samples)
+    cases = [
+        (tf2ss([1, 1], [1, 1.5, 1]), 0.25, 10),
+        (tf2ss([1, 2], [1, 1.5, 3, 1]), 0.5, 10),
+        (tf2ss(0.1, [1, 1, 2, 1]), 0.5, 10),
+    ]
+
+    # Some options for lsim2 and derived routines
+    tolerances = {'rtol': 1e-9, 'atol': 1e-11}
+
+    # Check that systems discretized with the impulse-invariant
+    # method really hold the invariant
+    @pytest.mark.parametrize("sys,sample_time,samples_number", cases)
+    def test_impulse_invariant(self, sys, sample_time, samples_number):
+        time = np.arange(samples_number) * sample_time
+        _, yout_cont = impulse2(sys, T=time, **self.tolerances)
+        _, yout_disc = dimpulse(c2d(sys, sample_time, method='impulse'), n=len(time))
```

Comment:
Sorry I missed this one PEP8 also complains about this one. By the way, you can install `pycodestyle` and test for PEP8 issues locally

Reply:
OK, added `pycodestyle` to my tools. Before that I used PyCharm's integrated linter, it doesn't bother about lines less that 120.